vcpkg failed to install azure-core-cpp on macOS

```
% ./vcpkg install azure-storage-blobs-cpp
Computing installation plan...
The following packages will be built and installed:
  * azure-core-cpp[core,curl,http]:x64-osx -> 1.0.0-beta.5
    azure-storage-blobs-cpp[core]:x64-osx -> 12.0.0-beta.7
  * azure-storage-common-cpp[core]:x64-osx -> 12.0.0-beta.7
  * curl[core,non-http,sectransp,ssl]:x64-osx -> 7.74.0#2
  * libiconv[core]:x64-osx -> 1.16#8
  * liblzma[core]:x64-osx -> 5.2.5#2
  * libxml2[core]:x64-osx -> 2.9.10#2
  * openssl[core]:x64-osx -> 1.1.1i
  * zlib[core]:x64-osx -> 1.2.11#9
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-osx...
Using cached binary package: /Users/jamis/.cache/vcpkg/archives/ee/eec560894eadc96bf595c09003ec4b02f6b5e566.zip
Using cached binary package: /Users/jamis/.cache/vcpkg/archives/be/be5e36e209a023a5cf5d6f7600ded9a3afc9e9e8.zip
Could not locate cached archive: /Users/jamis/.cache/vcpkg/archives/24/24202a978aa51a4cb592c626b98fb5ae4c09ba9c.zip
Could not locate cached archive: /Users/jamis/.cache/vcpkg/archives/12/12f206c65097a6483de628191b804d56201b783e.zip
Could not locate cached archive: /Users/jamis/.cache/vcpkg/archives/ca/ca61b9756db0aa27cbfd55a79fc94c408e5ef1eb.zip
Could not locate cached archive: /Users/jamis/.cache/vcpkg/archives/25/2509799e131255ef0117c71fef570cd608a52d78.zip
Could not locate cached archive: /Users/jamis/.cache/vcpkg/archives/49/495d424ce601701267e7dc93ec982cfbc86ba569.zip
Could not locate cached archive: /Users/jamis/.cache/vcpkg/archives/c2/c25934a254ae335df40db44e487e0e3833f9ecee.zip
Could not locate cached archive: /Users/jamis/.cache/vcpkg/archives/a0/a0f1c7cc01d826dd913457d07449877f74bb8638.zip
Starting package 1/9: zlib:x64-osx
Building package zlib[core]:x64-osx...
Building package zlib[core]:x64-osx... done
Installing package zlib[core]:x64-osx...
Installing package zlib[core]:x64-osx... done
Elapsed time for package zlib:x64-osx: 6.604 ms
Starting package 2/9: curl:x64-osx
Building package curl[core,non-http,sectransp,ssl]:x64-osx...
Building package curl[core,non-http,sectransp,ssl]:x64-osx... done
Installing package curl[core,non-http,sectransp,ssl]:x64-osx...
Installing package curl[core,non-http,sectransp,ssl]:x64-osx... done
Elapsed time for package curl:x64-osx: 14.11 ms
Starting package 3/9: azure-core-cpp:x64-osx
Building package azure-core-cpp[core,curl,http]:x64-osx...
-- Downloading https://github.com/Azure/azure-sdk-for-cpp/archive/azure-core_1.0.0-beta.5.tar.gz -> Azure-azure-sdk-for-cpp-azure-core_1.0.0-beta.5.tar.gz...
-- Extracting source /Users/jamis/Desktop/vcpkg/downloads/Azure-azure-sdk-for-cpp-azure-core_1.0.0-beta.5.tar.gz
-- Using source at /Users/jamis/Desktop/vcpkg/buildtrees/azure-core-cpp/src/0.0-beta.5-801f571f12.clean
-- Configuring x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:108 (message):
    Command failed: /Users/jamis/Desktop/vcpkg/downloads/tools/cmake-3.19.2-osx/cmake-3.19.2-macos-universal/CMake.app/Contents/bin/cmake /Users/jamis/Desktop/vcpkg/buildtrees/azure-core-cpp/src/0.0-beta.5-801f571f12.clean/sdk/core/azure-core/ -DBUILD_TRANSPORT_CURL=ON -DBUILD_TRANSPORT_WINHTTP=OFF -DWARNINGS_AS_ERRORS=OFF -DCMAKE_MAKE_PROGRAM=/Users/jamis/Desktop/vcpkg/downloads/tools/ninja-1.10.1-osx/ninja -DCMAKE_SYSTEM_NAME=Darwin -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/jamis/Desktop/vcpkg/scripts/toolchains/osx.cmake -DVCPKG_TARGET_TRIPLET=x64-osx -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/Users/jamis/Desktop/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/Users/jamis/Desktop/vcpkg -D_VCPKG_INSTALLED_DIR=/Users/jamis/Desktop/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF -DCMAKE_OSX_ARCHITECTURES=x86_64 -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/Users/jamis/Desktop/vcpkg/packages/azure-core-cpp_x64-osx/debug
    Working Directory: /Users/jamis/Desktop/vcpkg/buildtrees/azure-core-cpp/x64-osx-dbg
    Error code: 1
    See logs for more information:
      /Users/jamis/Desktop/vcpkg/buildtrees/azure-core-cpp/config-x64-osx-dbg-out.log
      /Users/jamis/Desktop/vcpkg/buildtrees/azure-core-cpp/config-x64-osx-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:328 (vcpkg_execute_required_process)
  ports/azure-core-cpp/portfile.cmake:15 (vcpkg_configure_cmake)
  scripts/ports.cmake:128 (include)


Error: Building package azure-core-cpp:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: azure-core-cpp:x64-osx
  Vcpkg version: 2021-01-13-unknownhash

Additionally, attach any relevant sections from the log files above.
```


in `vcpkg/buildtrees/azure-core-cpp/config-x64-osx-dbg-err.log`

```
By default, if no option is selected, on Unix, CURL transport adapter is used.
Libcurl version 7.74.0-DEV
Find the OpenSSL package on MacOS, used for Base64 encoding.
CMake Error at /Users/jamis/Desktop/vcpkg/downloads/tools/cmake-3.19.2-osx/cmake-3.19.2-macos-universal/CMake.app/Contents/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY
  OPENSSL_INCLUDE_DIR)
Call Stack (most recent call first):
  /Users/jamis/Desktop/vcpkg/downloads/tools/cmake-3.19.2-osx/cmake-3.19.2-macos-universal/CMake.app/Contents/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:582 (_FPHSA_FAILURE_MESSAGE)
  /Users/jamis/Desktop/vcpkg/downloads/tools/cmake-3.19.2-osx/cmake-3.19.2-macos-universal/CMake.app/Contents/share/cmake-3.19/Modules/FindOpenSSL.cmake:536 (find_package_handle_standard_args)
  /Users/jamis/Desktop/vcpkg/scripts/buildsystems/vcpkg.cmake:653 (_find_package)
  CMakeLists.txt:121 (find_package)

```

--- a/sdk/core/azure-core/CMakeLists.txt
+++ b/sdk/core/azure-core/CMakeLists.txt
@@ -116,10 +116,9 @@ create_code_coverage(core azure-core azure-core-test)
 
 target_link_libraries(azure-core INTERFACE Threads::Threads)
 
-if(MSVC)
+if(WIN32)
     target_link_libraries(azure-core PRIVATE crypt32)
 else()
-    message("Find the OpenSSL package on MacOS, used for Base64 encoding.")
     find_package(OpenSSL REQUIRED)
     target_link_libraries(azure-core PRIVATE OpenSSL::SSL)
 endif()

--- a/sdk/core/azure-core/vcpkg/CONTROL
+++ b/sdk/core/azure-core/vcpkg/CONTROL
@@ -3,6 +3,7 @@
 #
 Source: azure-core-cpp
 Version: @AZ_LIBRARY_VERSION@
+Build-Depends: openssl (!windows)
 Description: Microsoft Azure Core SDK for C++
   This library provides shared primitives, abstractions, and helpers for modern Azure SDK client libraries written in the C++.
 Homepage: https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/core/azure-core

--- a/sdk/core/azure-core/vcpkg/CONTROL
+++ b/sdk/core/azure-core/vcpkg/CONTROL
@@ -3,7 +3,7 @@
 #
 Source: azure-core-cpp
 Version: @AZ_LIBRARY_VERSION@
-Build-Depends: openssl (!windows)
+Build-Depends: openssl (!windows&!uwp)
 Description: Microsoft Azure Core SDK for C++
   This library provides shared primitives, abstractions, and helpers for modern Azure SDK client libraries written in the C++.
 Homepage: https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/core/azure-core

--- a/sdk/core/azure-core/vcpkg/Config.cmake.in
+++ b/sdk/core/azure-core/vcpkg/Config.cmake.in
@@ -5,8 +5,13 @@
 
 include(CMakeFindDependencyMacro)
 find_dependency(Threads)
+
 if(@BUILD_TRANSPORT_CURL@)
   find_dependency(CURL @CURL_MIN_REQUIRED_VERSION@)
+endif()
+
+if(NOT WIN32)
+  find_dependency(OpenSSL)
 endif()
 
 include("${CMAKE_CURRENT_LIST_DIR}/azure-core-cppTargets.cmake")

--- a/sdk/keyvault/azure-security-keyvault-common/vcpkg/CONTROL
+++ b/sdk/keyvault/azure-security-keyvault-common/vcpkg/CONTROL
@@ -1,6 +1,6 @@
 Source: azure-security-keyvault-common-cpp
 Version: @AZ_LIBRARY_VERSION@
-Build-Depends: azure-core-cpp, openssl (!windows)
+Build-Depends: azure-core-cpp
 Description: Microsoft Azure Common Key Vault SDK for C++
   This library provides common Azure KeyVault-related abstractions for Azure SDK.
 Homepage: https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/keyvault/azure-security-keyvault-common

--- a/sdk/keyvault/azure-security-keyvault-common/vcpkg/Config.cmake.in
+++ b/sdk/keyvault/azure-security-keyvault-common/vcpkg/Config.cmake.in
@@ -6,10 +6,6 @@
 include(CMakeFindDependencyMacro)
 find_dependency(azure-core-cpp)
 
-if(NOT MSVC)
-  find_dependency(OpenSSL)
-endif()
-
 include("${CMAKE_CURRENT_LIST_DIR}/azure-security-keyvault-common-cppTargets.cmake")
 
 check_required_components("azure-security-keyvault-common-cpp")

--- a/sdk/storage/azure-storage-common/CMakeLists.txt
+++ b/sdk/storage/azure-storage-common/CMakeLists.txt
@@ -80,7 +80,7 @@ target_link_libraries(azure-storage-common PUBLIC Azure::azure-core)
 target_include_directories(azure-storage-common PRIVATE ${LIBXML2_INCLUDE_DIRS})
 target_link_libraries(azure-storage-common PRIVATE ${LIBXML2_LIBRARIES})
 
-if(MSVC)
+if(WIN32)
     target_link_libraries(azure-storage-common PRIVATE bcrypt)
     # C28020 and C28204 are introduced by nlohmann/json
     target_compile_options(azure-storage-common PUBLIC /wd28204 /wd28020)

--- a/sdk/storage/azure-storage-common/vcpkg/Config.cmake.in
+++ b/sdk/storage/azure-storage-common/vcpkg/Config.cmake.in
@@ -8,7 +8,7 @@ find_dependency(LibXml2)
 find_dependency(Threads)
 find_dependency(azure-core-cpp)
 
-if(NOT MSVC)
+if(NOT WIN32)
   find_dependency(OpenSSL)
 endif()
 

